File: mvn-advent-of-code/src/test/java/advent/twenty_eighteen/Day12SubterraneanSustainabilityTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day12SubterraneanSustainabilityTest {
    @Test
    public void checkReal() {
        Day12SubterraneanSustainability im = new Day12SubterraneanSustainability("/twenty_eighteen/Day12-SubterraneanSustainability-input.txt");

        assertEquals(1816, im.getPart1Answer().longValue());
        assertEquals(399999999957l, im.getPart2Answer().longValue());
    }

    @Test
    public void checkExample1() {
        Day12SubterraneanSustainability im = new Day12SubterraneanSustainability("/twenty_eighteen/Day12-SubterraneanSustainability-example#1.txt");

        assertEquals(325, im.getPart1Answer().longValue());
        assertEquals(50000000501l, im.getPart2Answer().longValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_eighteen/Day12SubterraneanSustainabilityTest.scala
package advent.twenty_eighteen

import org.scalatest.{FlatSpec, Matchers}

class Day12SubterraneanSustainabilityTest  extends FlatSpec with Matchers {
  "2018 Day 12 - Input" should "calculate answers" in {
    val dp = new Day12SubterraneanSustainability("/twenty_eighteen/Day12-SubterraneanSustainability-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day12SubterraneanSustainability("/twenty_eighteen/Day12-SubterraneanSustainability-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
